File: Python/underautomation/universal_robots/xml_rpc/xml_rpc_unknown_value.py
import typing
from underautomation.universal_robots.xml_rpc.xml_rpc_type import XmlRpcType
from underautomation.universal_robots.xml_rpc.xml_rpc_value import XmlRpcValue
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.XmlRpc import XmlRpcUnknownValue as xml_rpc_unknown_value

class XmlRpcUnknownValue(XmlRpcValue):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = xml_rpc_unknown_value()
		else:
			self._instance = _internal
	@property
	def type(self) -> XmlRpcType:
		return XmlRpcType(self._instance.Type)
	@property
	def additional_information(self) -> str:
		return self._instance.AdditionalInformation

File: UnderAutomation.UniversalRobots.ObfuscatedSources/XmlRpc/XmlRpcUnknownValue.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.


namespace XmlRpc {
	/// <summary>
	/// Represents an unknown XML-RPC argument that has been received from the robot
	/// You can decode it yourself with the XML property
	/// </summary>
	public class XmlRpcUnknownValue : XmlRpcValue {

		/// <summary>
		/// Additional information about why this value could not be decoded.
		/// </summary>
		public readonly string AdditionalInformation;

		/// <summary>
		/// Returns the additional information about this unknown value.
		/// </summary>
		public override string ToString()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Determines whether the specified object is equal to this instance.
		/// </summary>
		/// <param name="obj">The object to compare with.</param>
		/// <returns><code>true</code> if the specified object is an <xref href="UnderAutomation.UniversalRobots.XmlRpc.XmlRpcUnknownValue" data-throw-if-not-resolved="false"></xref> with the same additional information; otherwise, <code>false</code>.</returns>
		public override bool Equals(object obj)
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Returns a hash code for this instance.
		/// </summary>
		public override int GetHashCode()
		{
			// Source is hidden, a Source licence is needed to access internal code...
			return default;
		}

		/// <summary>
		/// Gets the XML-RPC type of this value.
		/// </summary>
		public override XmlRpcType Type => default;
	}
}
